include/onnxruntime/core/session/environment.h

```diff
@@ -29,13 +32,20 @@ class Environment {
      Returns whether any runtime environment instance has been initialized.
   */
   static bool IsInitialized() { return is_initialized_; }
+  concurrency::ThreadPool* GetDefaultThreadPool() {
+    std::call_once(tp_once_, [this]() {
+      tp_ = new concurrency::ThreadPool("default", std::max<int>(std::thread::hardware_concurrency() - 1, 1));
```

Comment:
The previous version used std::thread::hardware_concurrency() / 2. Any reason we changed this? I think @skottmckay had a similar comment.

Reply:
Also, it looks like the threadpool is always a non-null ptr. Do we need to use a ptr then to pass it around? Can we use a reference instead? A ptr might imply that we need to check for null.

Reply:
Should this be called GetThreadPoolFromEnv? reason: the default threadpool is the one that is going to be supplied by the InferenceSession class if there's none provided by the env. The env's threadpool is not really the default one.

Reply:
I didn't want to change it. @skottmckay suggested me changing it, if I understood correctly.

Reply:
It could be NULL.

Reply:
> Should this be called GetThreadPoolFromEnv

No. This function is in the Env class. I don't think we need to rename it.

Reply:
> > Should this be called GetThreadPoolFromEnv
> 
> No. This function is in the Env class. I don't think we need to rename it.

The meaning of 'default' is confusing. What does default mean - the one that comes from the env or the one that the session class creates?

Reply:
Removed the "default" string.